### PR TITLE
Finish create word-answers

### PR DESCRIPTION
**Pull request by hitshootnum1**

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -22,4 +22,19 @@ def words_category category
   def content_category category_counter, category
     category_counter % 2 == 0 ? image_category(category) : words_category(category)
   end
+
+  def link_to_remove_fields(name, f)
```
Comment (hellovietnam93) [inline app/helpers/application_helper.rb]: remove ()

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -22,4 +22,19 @@ def words_category category
...
+    f.hidden_field(:_destroy) + link_to(name, "#",
+      onclick: "remove_fields(this); return false")
+  end
+
+  def link_to_add_fields(name, f, association)
```
Comment (hellovietnam93) [inline app/helpers/application_helper.rb]: remove ()

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -22,4 +22,19 @@ def words_category category
...
+    new_object = f.object.class.reflect_on_association(association).klass.new
+    fields = f.fields_for(association, new_object,
+      :child_index => "new_#{association}") do |builder|
+        render(association.to_s.singularize + "_fields", :f => builder)
```
Comment (hellovietnam93) [inline app/helpers/application_helper.rb]: render association.to_s.singularize + "_fields", f: builder

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -22,4 +22,19 @@ def words_category category
...
+
+  def link_to_remove_fields(name, f)
+    f.hidden_field(:_destroy) + link_to(name, "#",
+      onclick: "remove_fields(this); return false")
+  end
+
+  def link_to_add_fields(name, f, association)
+    new_object = f.object.class.reflect_on_association(association).klass.new
+    fields = f.fields_for(association, new_object,
+      :child_index => "new_#{association}") do |builder|
```
Comment (hellovietnam93) [inline app/helpers/application_helper.rb]: convention

File: app/models/word.rb [hunk 5]
```diff
@@ -2,6 +2,12 @@ class Word < ActiveRecord::Base
   belongs_to :cagetory
 
   has_many :lesson_words
-  has_many :word_answers
+  has_many :answers, class_name: "WordAnswer"
   has_many :lessons, through: :lesson_words
+
+  accepts_nested_attributes_for :answers, :reject_if => lambda {
+    |a| a[:content].blank?}, :allow_destroy => true
```
Comment (hellovietnam93) [inline app/models/word.rb]: convention

File: app/views/admin/words/_answer_fields.html.erb [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+      <td><%= f.text_field :content, class: "form-control" %></td>
+      <td>
+        <%= t "word.correct" %>
+        <%= f.check_box :correct, class: "correct" %>
+      </td>
+      <td><%= link_to_remove_fields t("word.remove"), f %></td>
+    </tr>
+  </table>
+<% end %>
+
```
Comment (hellovietnam93) [inline app/views/admin/words/_answer_fields.html.erb]: remove this line

File: app/models/word.rb [hunk 7]
```diff
@@ -2,6 +2,12 @@ class Word < ActiveRecord::Base
   belongs_to :cagetory
 
   has_many :lesson_words
-  has_many :word_answers
+  has_many :answers, class_name: "WordAnswer"
```
Comment (dieunb) [inline app/models/word.rb]: ` has_many :answers, class_name: WordAnswer.name`

File: app/views/admin/words/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,24 @@
+<h1><%= t "word.new" %></h1>
+<div class="row">
+  <%= form_for [:admin, @word] do |f| %>
+    <%= render "shared/error_messages", @word %>
+    <div class="col-md-4">
+      <h2><%= t "category.choose" %></h2>
+      <%= collection_select(:word, :category_id, Category.all, :id, :name, prompt: true) %>
```
Comment (hellovietnam93) [inline app/views/admin/words/new.html.erb]: Category.all register as an instance variable in controller

Comment (dieunb) [inline app/views/admin/words/new.html.erb]: assign `Category.all` to variable in controller

File: app/helpers/application_helper.rb [hunk 9]
```diff
@@ -22,4 +22,19 @@ def words_category category
...
+  def link_to_remove_fields name, f
+    f.hidden_field(:_destroy) + link_to(name, "#",
+      onclick: "remove_fields(this); return false")
+  end
+
+  def link_to_add_fields name, f, association
+    new_object = f.object.class.reflect_on_association(association).klass.new
+    fields = f.fields_for(association, new_object,
+      child_index: "new_#{association}") do |builder|
+      render(association.to_s.singularize + "_fields", f: builder)
```
Comment (hellovietnam93) [inline app/helpers/application_helper.rb]: remove ()

File: app/views/admin/words/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,24 @@
+<h1><%= t "word.new" %></h1>
+<div class="row">
+  <%= form_for [:admin, @word] do |f| %>
+    <%= render "shared/error_messages", object: @word %>
+    <div class="col-md-4">
+      <h2><%= t "category.choose" %></h2>
+      <%= collection_select(:word, :category_id, @categories, :id, :name, prompt: true) %>
```
Comment (hellovietnam93) [inline app/views/admin/words/new.html.erb]: remove ()

File: app/controllers/admin/words_controller.rb [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    else
+      flash[:danger] = t "word.not_create"
+      redirect_to root_path
+    end
+  end
+
+  private
+  def word_params
+    params.require(:word).permit :category_id, :content,
+      answers_attributes: [:content, :correct]
```
Comment (hellovietnam93) [inline app/controllers/admin/words_controller.rb]: it seem to be not enough

File: app/models/word_answer.rb [hunk 12]
```diff
@@ -1,5 +1,5 @@
 class WordAnswer < ActiveRecord::Base
-  belongs_to :user
+  belongs_to :word
 
   has_many :lesson_word
```
Comment (tqtrung09) [inline app/models/word_answer.rb]: use has_many, must  lesson_words

File: app/controllers/admin/words_controller.rb [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+
+  def create
+    @word = Word.new word_params
+    if @word.save
+      flash[:success] = t "word.create"
+      redirect_to root_path
+    else
+      flash[:danger] = t "word.not_create"
+      redirect_to root_path
+    end
```
Comment (tqtrung09) [inline app/controllers/admin/words_controller.rb]: Why redirect_to root_path  2 times ?

**Pull request merged**